File: fetch-function-happy.c
#include <stdio.h>
// Given an argument which is a file, fetch it, read it and print it

int fetch(char* fname) {
    // open fname
    FILE *fp = fopen(fname, "r");
    if (!fp) return 1;

    // read from fname and print
    char con[100];                          // store content in a char array
    while (fgets(con, 100, fp) != NULL)     // read from file into array
        printf("%s", con);                  // print content

    // close fname
    fclose(fp);
    return 0;
}

int main() {
    char *fname = "hello-function.c";
    return fetch(fname);
}

/*
# This: fetch-function-happy.c
# Prev: hello-function.c
# Next: fetch-function-error-handling.c

# Build & run:
gcc fetch-function.c -o fetch && ./fetch
*/

File: java/FetchFunctionHappy.java
import java.io.FileInputStream;

public class FetchFunctionHappy {

    private static void fetch(String fname) {
        try{
            FileInputStream fis = new FileInputStream(fname);
            int i;
            byte buf[] = new byte[100];
            while( (i = fis.read(buf))>0 ){
                String str = new String(buf,0,i);
                System.out.print(str);
            }
            fis.close();
        }catch( Exception e ){
            System.out.println(e);
        }
    }

    public static void main(String[] args) {
        String fname = "HelloFunction.java";
        fetch(fname);
    }
}

/*
# This: FetchFunctionHappy.java
# Prev: HelloFunction.java
# Next: -

# Build & run:
JNAME=FetchFunctionHappy; javac ${JNAME}.java && java ${JNAME}
*/

